src/sync_backend.rs

```diff
@@ -64,8 +64,17 @@ impl<'a> SyncBackend for RobloxSyncBackend<'a> {
 
             Err(RobloxApiError::ResponseError {
                 status: StatusCode::TOO_MANY_REQUESTS,
+                headers,
                 ..
-            }) => Err(Error::RateLimited),
+            }) => Err(Error::RateLimited {
+                wait_seconds: headers
+                    .get("retry-after")
+                    .ok_or(RobloxApiError::MissingRetryHeader)?
```

Comment:
Instead of erroring when the header is missing or panicking when invalid, it would be better to just default it to what the user has provided